src/main/scala/com/tuplejump/kafka/connect/cassandra/package.scala

```diff
@@ -69,29 +71,85 @@ package object cassandra {
 
   implicit class SinkRecordOps(record: SinkRecord) {
 
-    def as(namespace: String): SinkQuery = {
+    def as(namespace: String, fieldMapping: Map[String, Any]): SinkQuery = {
+      val colNamesVsValues: Map[String, String] = {
+        if (fieldMapping.isEmpty) {
+          convertToCqlData
+        } else {
+          convertToCqlData(fieldMapping)
+        }
+      }
+      SinkQuery(namespace, colNamesVsValues)
+    }
+
+    def convertToCqlData(): (Map[String, String]) = {
       val schema = record.valueSchema
-      val columnNames = schema.asColumnNames
-      val columnValues = schema.`type`() match {
+      schema.`type`() match {
+        case STRUCT =>
+          val struct: Struct = record.value.asInstanceOf[Struct]
+          schema.fields.asScala.map { field =>
+            (field.name, convert(schema, struct, field))
+          }.toMap
+        case other =>
+          throw new DataException(
+            s"Unable to create insert statement with unsupported value schema type $other.")
+      }
+    }
+
+    def convertToCqlData(fieldMapping: Map[String, Any]): Map[String, String] = {
+      lazy val exception = new DataException(s"Invalid fieldMapping received - $fieldMapping")
+      val result = scala.collection.mutable.Map.empty[String, String]
+      val valueSchema = record.valueSchema
+      valueSchema.`type`() match {
         case STRUCT =>
           val struct: Struct = record.value.asInstanceOf[Struct]
-          columnNames.map(convert(schema, struct, _)).mkString(",")
-        case other => throw new DataException(
-          s"Unable to create insert statement with unsupported value schema type $other.")
+          populateResultsMap(result, struct, fieldMapping.asInstanceOf[Map[String, Any]])
```

Comment:
You don't ever need to do this 'asInstanceOf', it is always guaranteed to be that type after load time configuration.